### DISCUSSION TITLE
React in account client

### DISCUSSION THREAD

**Discussion by ahcamy2021** [keycloak/keycloak#11549] [Q&A] [answered]
Hi,

I saw that the version of React used in account client is 16.9, I want to know if there is any plan to upgrade the version of React in the future versions of KC?

Thanks in adavance.

**Comment by ssilvert** [comment, marked as answer]
The account UI has been neglected due to the work on the new admin UI, but we definitely plan to put a lot of work into the account UI later this year.  The React version will definitely get bumped to the latest.  

Also we plan to move it to the keycloak-ui repo where we can keep a better eye on it.  So we will be doing a better job of keeping it up to date in the future.

**Reply by xgp** [reply]
Is there an updated roadmap or dates for this transition? I'm trying to figure out what the recommended way to build a custom account theme that will (hopefully) be the recommendation for a year or two. Is the plan to just port the `keycloak.v2` theme to use vite in the new repo, or is it going to be something fundamentally different? Thank you!

**Comment by ssilvert** [comment]
@xgp It will be very similar to the way we built the new admin console.  Yes, it will use vite.

We have started working on it in fits and starts.  You can see what has been done here: https://github.com/keycloak/keycloak-ui/tree/main/apps/account-ui

Now we have decided to move the keycloak/keycloak-ui repo back to keycloak/keycloak.  I think that it won't take long to get a release of the vite version of account console completed shortly thereafter.  But it's hard to give a timeline right now.

**Comment by xgp** [comment]
@ssilvert Appreciate the answer. This is exactly what I was going for.
